File: game-of-go-angular/src/app/common/game-state.ts
import { GameResult } from "./game-result";

export class GameState {
    gameBoard?: number[];
    log?: string;
    blackCaptures?: number;
    whiteCaptures?: number;
    lastColor?: number;
    lastMove?: string;
    nextColor?: number;
    result?: GameResult;
    chatLog?: string[];
}

File: game-of-go-spring-boot/src/main/java/io/github/aylesw/igo/game/GameState.java
package io.github.aylesw.igo.game;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class GameState {
    int[] gameBoard;
    String log;
    int blackCaptures;
    int whiteCaptures;
    int lastColor;
    String lastMove;
    int nextColor;
    List<String> chatLog;
}
